src/lib/coda_networking/coda_networking.ml

```diff
@@ -557,32 +681,28 @@ let wrap_rpc_data_in_envelope conn data =
 
 let create (config : Config.t)
     ~(get_staged_ledger_aux_and_pending_coinbases_at_hash :
-          State_hash.t Envelope.Incoming.t
-       -> (Staged_ledger.Scan_state.t * Ledger_hash.t * Pending_coinbase.t)
-          option
+          Rpcs.Get_staged_ledger_aux_and_pending_coinbases_at_hash.query
+          Envelope.Incoming.t
+       -> Rpcs.Get_staged_ledger_aux_and_pending_coinbases_at_hash.response
           Deferred.t)
     ~(answer_sync_ledger_query :
-          (Ledger_hash.t * Ledger.Location.Addr.t Syncable_ledger.Query.t)
-          Envelope.Incoming.t
-       -> Sync_ledger.Answer.t Deferred.Or_error.t)
+          Rpcs.Answer_sync_ledger_query.query Envelope.Incoming.t
+       -> Rpcs.Answer_sync_ledger_query.response Deferred.t)
     ~(get_ancestry :
-          Consensus.Data.Consensus_state.Value.t Envelope.Incoming.t
-       -> ( External_transition.t
-          , State_body_hash.t list * External_transition.t )
-          Proof_carrying_data.t
-          Deferred.Option.t)
+          Rpcs.Get_ancestry.query Envelope.Incoming.t
+       -> Rpcs.Get_ancestry.response Deferred.t)
     ~(get_best_tip :
-          unit Envelope.Incoming.t
-       -> ( External_transition.t
-          , State_body_hash.t list * External_transition.t )
-          Proof_carrying_data.t
-          Deferred.Option.t)
+          Rpcs.Get_best_tip.query Envelope.Incoming.t
+       -> Rpcs.Get_best_tip.response Deferred.t)
+    ~(get_telemetry_data :
+          Rpcs.Get_telemetry_data.query Envelope.Incoming.t
+       -> Rpcs.Get_telemetry_data.response Deferred.t)
```

Comment:
You would be able to avoid the ugly `ref None` knot tying hack where you instantiate the code networking implementations if you changed this argument to the following type:

```ocaml
   t
-> Rpcs.Get_telemetry_data.query Envelope.Incoming.t Deferred.Option.t)	       -> Rpcs.Get_telemetry_data.response Deferred.t
```

Reply:
I'm not sure I follow: when you're setting up `get_telemetry_data_rpc` in `Coda_networking`, you don't yet have a `Coda_networking.t` to input.

I'll merge this for now, and we can make this change later.